source/Xunit.Gherkin.Quick/Feature.cs

```diff
@@ -45,13 +45,27 @@ internal void Scenario(string scenarioName)
                 object[] methodParamValues = null;
                 if (methodParams.Length > 0)
                 {
-                    var methodParamStringValues = stepRegexMatch.Groups.Cast<Group>().Skip(1).Select(g => g.Value).ToList();
+                    var methodParamStringValues = stepRegexMatch.Groups.Cast<Group>().Skip(1).Select(g => g.Value).ToList<object>();
 
-                    if (methodParamStringValues.Count < methodParams.Length)
+                    if (methodParamStringValues.Count < methodParams.Length && parsedStep.Argument == null)
+                    {
                         throw new InvalidOperationException($"Method `{matchingStepMethod.method.Name}` for step `{parsedStep.Keyword}{parsedStep.Text}` is expecting {methodParams.Length} params, but only {methodParamStringValues.Count} param values were supplied.");
+                    }
+
+                    // A step argument (Table/Multiline String, etc.) is always last is the list of arguments
+                    if (parsedStep.Argument != null)
+                        methodParamStringValues.Add(parsedStep.Argument);
 
-                    methodParamValues = methodParams.Select((p, i) => Convert.ChangeType(methodParamStringValues[i], p.ParameterType))
-                        .ToArray();
+                    methodParamValues = methodParams
+                        .Select((p, i) => {
+                            if (methodParamStringValues[i].GetType() == typeof(DataTable)) // DataTable support
+                                return new Table(((DataTable)methodParamStringValues[i]).Rows.ToArray());
+                            if (methodParamStringValues[i].GetType() == typeof(DocString)) // Multiline string
+                                return (((DocString)methodParamStringValues[i]).Content); 
```

Comment:
Why not return DocString instead of returning just content?
DocString has another property too, which is content type, and maybe the method wants to examine that?

Reply:
I am now aware why SpecFlow decided to not give DocString directly. Especially that it has additional properties on it. I want to give ability to examine DocString. SpecFlow's design is not a guideline in this case.
Although, I think that we can support both string and DocString. But I'm not sure how that would be expressive at this point. So, phase 1 - let's support DocString. I see string a concrete case of DocString, so we can add that support later. But DocString support must be there first.

Reply:
In general, this framework is not supposed to be directly compatible with SpecFlow. Not so far. Even FeatureFile attribute is not present in SpecFlow.
I'm sorry if that was your impression. But I don't see a need to complicate a framework just to mimic SpecFlow. Lightweight nature is important to be maintained.

Reply:
Gotcha.  I didn't realize you don't use SpecFlow.  We're heavy SpecFlow users on our team, except for our .Net Core projects, which has been discouraging, because executable Gherkin documentation, particularly when post-processed via Pickles (http://www.picklesdoc.com/), really helps make our tests cleaner and more useful inside and out outside the dev team.  

I periodically look for .NET Core support in the SpecFlow project and ran across this library from this post (https://stackoverflow.com/questions/46935336/specflow-on-net-core-2-0).  You're project looked promising in that I could use it and not have to change much of what I do today with SpecFlow.  My hopes were, and I'd think it would apply to most people using C#, that this would be as backward compatible to SpecFlow as possible, as least for the primary use cases and allow us to continue to use the SpecFlow tooling, since it autogenerates the method stubs and colorizes / auto-formats our feature files.

If that's not where you're heading with this project that's cool, but at least you'll understand where I'm coming from.

Reply:
I understand.
So, the change from Table to DataTable should be minor probably, if Table derives from DataTable. So where SpecFlow compatible method has Table, just replace that with DataTable. It should continue working then.
As about DocString, as I said, I do see value in supporting both. But I don't want to support only string. But even if we don't support string at all, that's still a minor difference for the consumer code.

What I'm trying to say is - this framework is still close to SpecFlow, but promise is not to make it backwards compatible with SpecFlow. I want to keep the framework lightweight and clean from design standpoint. That's higher by priority than following SpecFlow design by any means.

Reply:
Given that, I don't feel like this library will work for me.  It's probably best that I wait for SpecFlow to finish their .NET Core support.  Thanks for the dialog.